draft-gpew-priv-ppm.md

```diff
@@ -1123,7 +1125,7 @@ the report sequence any report for which the initial state is FAILED.
 
 Let `[aggregator]` denote the helper's API endpoint. The leader extracts the
 sequence of report shares for `[aggregator]` and sends sends a POST request to
-`[aggregator]/aggregate_init` with an Aggregate message of type `agg_init_req`
+`[aggregator]/aggregate` with an Aggregate message of type `agg_init_req`
```

Comment:
I keep going back and forth on whether there should be a separate endpoints for `aggreagate_init`, `aggregate` and `aggregate_share`. I think there's a disconnect between having separate endpoints, but having them all take `struct Aggregate`. The thing is, the only valid message to send to `[helper]/aggregate_init` is an `AggregateInitReq`, and the only valid response to send is `AggregateResp`, so why do we have a wrapper that allows sending or responding with any of `AggregateReq`, `AggregateReq`, `AggregateResp`, `AggregateShareReq` or `HpkeCiphertext`? I think the reason is so that we can specify the `opaque tag[32]` for the HMAC-SHA256 in one place but I don't think it's worth the extra error cases introduced by the enum wrapper.

I think we should have:
- `[helper]/aggregate_init` which _only_ accepts `AggregateInitReq` and _only_ responds with `AggregateResp`
- `[helper]/aggregate` which _only_ accepts `AggregateReq` and _only_ responds with `AggregateResp`
- `[helper]/aggregate_share` which _only_ accepts `AggregateShareReq` and _only_ responds with `HpkeCiphertext`
- add `opaque tag [32]` fields to `AggregateInitReq`, `AggregateReq` and `AggregateShareReq` so that helper can verify that the message came from the leader.

Finally, here's something spicy: do we need the HMAC tag on `AggregateResp` or the `HpkeCiphertext` that the helper responds to the leader with? If the leader connects to helper over TLS, then that provides sufficient authentication for responses.